### DISCUSSION TITLE
Trying to launch the tests from Run and Debug, it is failing.

### DISCUSSION THREAD

**Discussion by ashish-gawali-adi** [redhat-developer/vscode-extension-tester#704] [Q&A] [answered]
Referring to the "helloworld-sample" and trying to launch the tests from the "Debug and Run" prompt getting the following error:
![image](https://user-images.githubusercontent.com/126178747/226707734-93645dd2-794f-47f4-b1a2-67255428aa41.png)

"${env:NODE_OPTIONS}='--require "c:/Users/AGawali/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/ms-vscode.js-debug/src/bootloader.bundle.js" --inspect-publish-uid=http'; ${env:VSCODE_INSPECTOR_OPTIONS}='{"inspectorIpc":"\\\\.\\pipe\\node-cdp.26888-1.sock","deferredMode":false,"waitForDebugger":"","execPath":"C:\\Program Files\\nodejs\\node.exe","onlyEntrypoint":false,"autoAttachMode":"always","fileCallback":"C:\\Users\\AGawali\\AppData\\Local\\Temp\\node-debug-callback-16d0ca29f2c00b4d"}'; & 'C:\Program Files\nodejs\node.exe' '.\node_modules\.bin\extest' 'setup-and-run' 'C:\workdir\trials\vscode-extension-tester\vscode-extension-tester\sample-projects\helloworld-sample/out/ui-test/*.js' '-o' 'settings.json' '--mocha_config' 'C:\workdir\trials\vscode-extension-tester\vscode-extension-tester\sample-projects\helloworld-sample/src/ui-test/.mocharc-debug.js'
Debugger attached.
Waiting for the debugger to disconnect...
C:\workdir\trials\vscode-extension-tester\vscode-extension-tester\sample-projects\helloworld-sample\node_modules\.bin\extest:2
basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")
          ^^^^^^^

SyntaxError: missing ) after argument list
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1027:15)
    at Module._compile (node:internal/modules/cjs/loader:1063:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47"

Node.js v17.9.1

**Comment by djelinek** [comment, marked as answer]
the sample project was upgraded and I have tested the launch configuration is working with NodeJS 18.x.x

**Comment by swathisuja** [comment]
I am executing commands individually from get-vscode, get-chromedrover, install-vsix and run-tests using API commands in my extension. 

Added debug config :
 {
            "name": "Debug UI Tests",
            "type": "node",
             "request": "launch",
            "program": "${workspaceFolder}/node_modules/vscode-extension-tester/out/cli.js",
            "args": [
                    "get-vscode",
                    "get-chromedriver",
                    "install-vsix",
                    "run-tests",                 
                    "${workspaceFolder}/dist/test/*-test.js",
                    "--mocha_config",
                    "${workspaceFolder}/src/test/.mocharc-debug.js"
                     ],
            "console": "integratedTerminal",
            "runtimeExecutable": "node",
             "internalConsoleOptions": "neverOpen"
        },

IS my debug configuration right? 

I am getting error: unknown option '--mocha_config'. 

Although -m --mocha_config is supported option for run-tests and .mocharc-debug.js was added in src/test directory
![image](https://github.com/redhat-developer/vscode-extension-tester/assets/28916423/44562478-e3c7-4951-9b7d-b0ba2ab45360)


Help me to set up debug tests.
